rfcs/0002-ckb/0002-ckb.md

```diff
@@ -7,372 +7,232 @@ Organization: Nervos Foundation
+Correctness includes eventual consistency that guarantees every node see identical copies of state and availability that makes sure the network respond to user’s requests within a reasonable time, and fairness that ensures mining nodes to get fair returns for their efforts to keep the network securely functioning.
 
-### 2.3  Transaction
+High performance includes transaction latency which is the time between the submission of a request and the confirmation of its execution results, and transaction throughput as the number of transactions the system is capable of processing per second. Both of them depends on block time, which is the average time between two consecutive blocks. CKB consensus improves both transaction latency and throughput by using network bandwidth more efficiently without sacrificing security and decentralization.
 
-Transactions express update and transfer of cells. In a single transaction, users can update data in one or more cells, or transfer cells to another user. A transaction includes the following:
+Please check the CKB Consensus Paper for more details.
 
-- deps: dependent cell set, provides read-only data that validation needs. They must be references to P1 cells or user input.
-- inputs: input cell set, includes cells to be transferred and/or updated. They must be references to P1 cells with corresponding unlock scripts.
-- outputs: output cell set, includes all newly created P1 cells.
+## 4. Programming Model
 
-Because cells are immutable, cells cannot be modified directly. Instead, new versions of cells are created and all cell versions can be linked together to form a “cell version chain”. A cell capacity transfer creates the cell’s first version, and the updates become its historical versions. The head of a cell version chain is its current version. The CKB is the superset of all cell version chains. The set of all cell heads is the current version of the CKB.
+CKB provides a stateful Turing-complete programming model based on CKB VM and cell model.
 
-The design of the CKB cell model and transactions is friendly to light clients. Since all the states are in blocks, block synchronization also accomplishes state synchronization. Light clients only need to synchronize blocks, and do not need to perform state transition computations. If we only stored events in blocks, we would have needed full nodes to also support state synchronization. This extra protocol can be difficult for large deployments, because the incentive to do so is not defined within the blockchain protocol. CKB defines state synchronization in the protocol itself, and this makes light nodes and full nodes more equal peers, leading to a more robust and decentralized system.
+| | Bitcoin | Ethereum | CKB |
+|-|---------|----------|------------|
+|Instruction Set|Script|EVM|RISC-V|
+|Cryptographic Primitive|Opcode|Precompile|Syscall|
```

Comment:
I think we haven't reached the point where we need to use syscall to implement crypto primitives, correct?

Reply:
Correct, 'Syscall' here is wrong. I was thinking of other things when writing this. Fixed.